=)

realbusyman opened the pull request:
Hello! First thank you for this great component!)
I have fixed a bug with items suddenly disappear and now your component is perfect!)
And I've translated (with little help of Google) comments to English.

Commit:
Comments translated to English (by Google)

Commit:
items suddenly disappear bug fixed

Commit:
highlight item fixed (after prev.fix)

Commit:
more accurate code around offscreen margin

Commit:
creating first element bad access fix

adow commented:
Hi,

Thanks for your fixing bugs in WKPagesCollection.
I have merged you request and the bug with items suddenly disappeare is gone. 
I am wondering how topOffScreenMargin(120) work? Why this collectionView should be higher than screen and why it should be placed at top (-120.0f)?
I have never used UICollectionView before and it really confused me.

I have changed `-(CGSize)collectionViewContentSize` in `WKPagesCollectionViewFlowLayout` and added topOffScreenMargin to the height or  the deletion animation would be incorrect.

在 2014年4月4日，下午11:23，Николай notifications@github.com 写道：

> Hello! First thank you for this great component!)
> I have fixed a bug with items suddenly disappear and now your component is perfect!)
> And I've translated (with little help of Google) comments to English.
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/realbusyman/WKPagesCollectionView master
> Or view, comment on, or merge it at:
> 
>   https://github.com/adow/WKPagesCollectionView/pull/2
> 
> Commit Summary
> 
> Comments translated to English (by Google)
> items suddenly disappear bug fixed
> File Changes
> 
> M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionView.h (22)
> M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionView.m (22)
> M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionViewCell.h (8)
> M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionViewCell.m (10)
> M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionViewFlowLayout.m (8)
> Patch Links:
> 
> https://github.com/adow/WKPagesCollectionView/pull/2.patch
> https://github.com/adow/WKPagesCollectionView/pull/2.diff
> —
> Reply to this email directly or view it on GitHub.

realbusyman commented:
Hi!
I think my decision is not flexible enough. We need to make topOffScreenMargin
calculated value instead of constant.
The magic is simple) It's because of memory management of UICollectionView.
CollectionView has to release part of cells when they gone off the screen.
In WKPagesCollectionView it releases 3 cells at once. One of these cells
always was at screen. So if we add margin all of these 3 cells will be out
of the screen when they has to release. I dont remember how I've calculate
this value but it has to be calculated using height of cell.

2014-04-16 18:49 GMT+04:00 adow notifications@github.com:

> Hi,
> 
> Thanks for your fixing bugs in WKPagesCollection.
> I have merged you request and the bug with items suddenly disappeare is
> gone.
> I am wondering how topOffScreenMargin(120) work? Why this collectionView
> should be higher than screen and why it should be placed at top (-120.0f)?
> I have never used UICollectionView before and it really confused me.
> 
> I have changed `-(CGSize)collectionViewContentSize` in
> `WKPagesCollectionViewFlowLayout` and added topOffScreenMargin to the
> height or the deletion animation would be incorrect.
> 
> 在 2014年4月4日，下午11:23，Николай notifications@github.com 写道：
> 
> > Hello! First thank you for this great component!)
> > I have fixed a bug with items suddenly disappear and now your component
> > is perfect!)
> > And I've translated (with little help of Google) comments to English.
> > 
> > You can merge this Pull Request by running
> > 
> > git pull https://github.com/realbusyman/WKPagesCollectionView master
> > Or view, comment on, or merge it at:
> > 
> > https://github.com/adow/WKPagesCollectionView/pull/2
> > 
> > Commit Summary
> > 
> > Comments translated to English (by Google)
> > items suddenly disappear bug fixed
> > File Changes
> > 
> > M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionView.h (22)
> > M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionView.m (22)
> > M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionViewCell.h
> > (8)
> > M WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionViewCell.m
> > (10)
> > M
> > WKPagesScrollView/WKPagesCollectionView/WKPagesCollectionViewFlowLayout.m
> > (8)
> > Patch Links:
> > 
> > https://github.com/adow/WKPagesCollectionView/pull/2.patch
> > https://github.com/adow/WKPagesCollectionView/pull/2.diff
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/adow/WKPagesCollectionView/pull/2#issuecomment-40607792
> .

## 

с уважением, Николай Абеляшев

adow commented:
Yes I have figured out the meaning of topOffScreenMargin. 
There is another bug because I added topOffScreenMargin to height in `-(CGSize)collectionConentSize`. https://github.com/adow/WKPagesCollectionView/commit/3b046e62900beb261c8fb86b2a534211818ad5e9
There is a big space between last item and the bottom of screen.

I have to add this value, otherwise animation would be incorrect if you delete last item by swipe gesture. 

![ios 2014417 9 42 54](https://cloud.githubusercontent.com/assets/504314/2727365/795c3068-c5d2-11e3-9564-ae39afeda82b.png)
